compiler/src/iree/compiler/Dialect/Flow/Transforms/CollapseReductionDims.cpp

```diff
@@ -31,7 +31,8 @@ static bool hasContiguousDims(AffineMap map, ArrayRef<unsigned> dims) {
     }
     // Check that the following dimensions are match the order of `dims`
     for (unsigned j = 1, numDims = dims.size(); j < numDims; j++) {
-      if (map.getDimPosition(i + j) != dims[j]) {
```

Comment:
This does seem like a bug.... We should be collapsing these dimensions. I am not fully sure I follow the details here, but would be great if you could file a bug with a repro (actually your test is a repro), and I can get Guray to take a look at it.

Reply:
Cool. Thanks Mahesh! The test input I have has a broadcast so we can't simply collapse. Nothing to collapse since it is 1D, so it is more like 1D -> broadcast -> collapse to 1D, but I don't know it is beneficial to materialize the broadcast in buffer.

Reply:
Filed https://github.com/iree-org/iree/issues/12211